common/workunit/workunit.cpp

```diff
@@ -13937,13 +13937,16 @@ extern WORKUNIT_API double getThorManagerRate()
 
 extern WORKUNIT_API double getThorWorkerRate()
 {
+    // Note: (bare-metal) the use of getAffinityCpus to get the number of CPUs used by workers
+    // doesn't really make sense since the caller is likely to be running on thor manager (so it will
+    // return cpu affinity for the manager, rather than for the worker).  This needs rethinking.
```

Comment:
agreed. Is there a JIRA to track this? (can you link it to HPCC-31569)

Reply:
done. I've created https://hpccsystems.atlassian.net/browse/HPCC-31572 and linked to this issue.